code/pages/SubsitesVirtualPage.php

```diff
@@ -170,24 +196,44 @@ public function onBeforeWrite()
             $this->ExtraMeta = $this->ContentSource()->ExtraMeta ? $this->ContentSource()->ExtraMeta : $this->ExtraMeta;
         }
     }
-}
 
-class SubsitesVirtualPage_Controller extends VirtualPage_Controller
-{
-    public function reloadContent()
+    public function validURLSegment()
     {
-        $this->failover->copyFrom($this->failover->CopyContentFrom());
-        $this->failover->write();
-        return;
-    }
+        $isValid = parent::validURLSegment();
+
+        // Veto the validation rules if its false. In this case, some logic
+        // needs to be duplicated from parent to find out the exact reason the validation failed.
+        if (!$isValid) {
+            $IDFilter = $this->ID ? "AND \"SiteTree\".\"ID\" <> $this->ID" : null;
+            $parentFilter = null;
+
+            if (Config::inst()->get(SiteTree::class, 'nested_urls')) {
+                if ($this->ParentID) {
+                    $parentFilter = " AND \"SiteTree\".\"ParentID\" = $this->ParentID";
+                } else {
+                    $parentFilter = ' AND "SiteTree"."ParentID" = 0';
+                }
+            }
 
-    public function init()
-    {
-        $origDisableSubsiteFilter = Subsite::$disable_subsite_filter;
-        Subsite::$disable_subsite_filter = true;
+            $origDisableSubsiteFilter = Subsite::$disable_subsite_filter;
+            Subsite::$disable_subsite_filter = true;
+            $existingPage = DataObject::get_one(
+                'SilverStripe\\CMS\\Model\\SiteTree',
```

Comment:
It might be nice to replace all of this with ORM filters, but for now I'd just say use `SiteTree::class` instead of this line, and below too